nexus/transport.go

```diff
@@ -0,0 +1,675 @@
+// Use TransportGetOperationResultResponse.GetResult to retrieve the final value or error returned by the operation.
+//
+// NOTE: Experimental
+type TransportGetOperationResultResponse[T any] struct {
+	result *OperationResult[T]
+	Links  []Link
+}
+
+// GetResult returns the final result or error returned by the operation.
+//
+// NOTE: Experimental
+func (gr *TransportGetOperationResultResponse[T]) GetResult() (T, error) {
+	return gr.result.Get()
+}
+
+// TransportCancelOperationResponse is the response to Transport.CancelOperation calls.
+//
+// NOTE: Experimental
+type TransportCancelOperationResponse struct{}
+
+// OperationResult contains the final value or error returned by an operation handler. One and only one of
+// result or err will be populated. Use OperationResult.Get to retrieve the result.
+//
+// In most cases err will be an [OperationError]. Other failures, such as [HandlerError], will be returned by
+// the Transport method called to indicate a failure to communicate with the operation handler.
+//
+// NOTE: Experimental
+type OperationResult[T any] struct {
+	result T
+	err    error
+}
+
+// Get returns the final result or error returned by an operation. Only one of result or err should be non-zero/non-nil.
+//
+// NOTE: Experimental
+func (r *OperationResult[T]) Get() (T, error) {
+	return r.result, r.err
+}
+
+// User-Agent header set on HTTP requests.
+const userAgent = "Nexus-go-sdk/" + version
+const headerUserAgent = "User-Agent"
+const getResultContextPadding = time.Second * 5
+
+var errOperationWaitTimeout = errors.New("operation wait timeout")
+
+type (
+	// HTTPTransport is a [Transport] implementation backed by HTTP.
+	//
+	// NOTE: Experimental
+	HTTPTransport struct {
+		options        HTTPTransportOptions
+		serviceBaseURL *url.URL
+	}
+
+	// HTTPTransportOptions are the options for constructing a new [HTTPTransport].
+	//
+	// NOTE: Experimental
+	HTTPTransportOptions struct {
+		// Base URL for all requests. Required.
+		BaseURL string
+		// A function for making HTTP requests.
+		// Defaults to [http.DefaultClient.Do].
+		HTTPCaller func(*http.Request) (*http.Response, error)
+		// A [Serializer] to customize client serialization behavior.
+		// By default the client handles JSONables, byte slices, and nil.
+		Serializer Serializer
+		// A [FailureConverter] to convert a [Failure] instance to and from an [error]. Defaults to
+		// [DefaultFailureConverter].
+		FailureConverter FailureConverter
+		// UseOperationID instructs the client to use an older format of the protocol where operation ID is sent
+		// as part of the URL path.
+		// This flag will be removed in a future release.
+		//
+		// NOTE: Experimental
+		UseOperationID bool
+	}
+)
+
+// NewHTTPTransport creates a new [Transport] backed by HTTP from the provided [HTTPTransportOptions].
+//
+// NOTE: Experimental
+func NewHTTPTransport(options HTTPTransportOptions) (*HTTPTransport, error) {
```

Comment:
Consider also adding a `NewHTTPClient` helper with `HTTPClientOptions` that flatten out all of the shared client and transport options.

Reply:
Okay, we can bring it back later.